bftengine/src/bftengine/SigManager.cpp

```diff
@@ -133,14 +133,20 @@ SigManager::SigManager(PrincipalId myId,
     ConcordAssert(p.first != myId_);
 
     auto iter = publicKeyIndexToVerifier.find(p.second);
+    const auto& [key, format] = publickeys[p.second];
     if (iter == publicKeyIndexToVerifier.end()) {
-      const auto& keyPair = publickeys[p.second];
-      verifiers_[p.first] = new RSAVerifier(keyPair.first.c_str(), keyPair.second);
+      verifiers_[p.first] = new RSAVerifier(key.c_str(), format);
       publicKeyIndexToVerifier[p.second] = verifiers_[p.first];
-    } else
+    } else {
       verifiers_[p.first] = iter->second;
+    }
+    if (replicasInfo_.isIdOfExternalClient(p.first)) {
+      clientsPublicKeys_.ids_to_keys[p.first] = concord::messages::keys_and_signatures::PublicKey{key, (uint8_t)format};
+      LOG_DEBUG(ON_CHAIN_LOG, "Adding key of client " << p.first);
+    }
   }
 
+  LOG_DEBUG(ON_CHAIN_LOG, "Map contains " << clientsPublicKeys_.ids_to_keys.size() << " public clients keys");
```

Comment:
@glevkovich can you confirm that this code, inserts correctly the public keys of the external clients?

Reply:
@cloudnoize , looks fine to me..

Reply:
Thanks for the verification @glevkovich !